### PR TITLE
Fix misleading error message when using optional or repeatedly parser

### PR DESCRIPTION
**Pull request by ashihaby**
#34

### CODE REVIEW

File: src/main/scala/scala/util/parsing/combinator/RegexParsers.scala [hunk 1]
```diff
@@ -137,8 +137,19 @@ trait RegexParsers extends Parsers {
...
+   * `phrase(p)` succeeds if `p` succeeds and no input is left over after `p`.
+   *
+   * @param p the parser that must consume all input for the resulting parser
+   *          to succeed.
+   *
+   * @return  a parser that has the same result as `p`, but that only succeeds
+   *          if `p` consumed all the input.
+   */
   override def phrase[T](p: Parser[T]): Parser[T] =
-    super.phrase(p <~ opt("""\z""".r))
```
Comment (ashihaby) [inline src/main/scala/scala/util/parsing/combinator/RegexParsers.scala]: @gourlaysama This line was added before to fix this [issue](https://issues.scala-lang.org/browse/SI-1178) to consume the trailing whitespace at the end of a `phrase` parser. But that produced a new issue that I have mentioned. I think the proper fix is to use regex that match nothing.

**Comment by adriaanm** [comment]
ping @gourlaysama

File: src/main/scala/scala/util/parsing/combinator/RegexParsers.scala (hunk 1 above)
Comment (gourlaysama) [inline src/main/scala/scala/util/parsing/combinator/RegexParsers.scala]: Good catch! That makes the error message much clearer, even when there is no `lastNoSuccessVar`. And as far as I can tell it behaves the same wrt Success/Failure results, so it probably can be backported to 1.0.x.

**Comment by gourlaysama** [comment]
@ashihaby sorry it took so long to get to this. Thank you for your contribution and perseverance :-)

**Pull request merged**